controller/app.go

```diff
@@ -73,21 +79,29 @@ func (r *AppRepo) Add(data interface{}) error {
 			Domain:  fmt.Sprintf("%s.%s", app.Name, r.defaultDomain),
 			Service: app.Name + "-web",
 		}).ToRoute()
-		route.ParentRef = routeParentRef(app.ID)
-		if err := r.router.CreateRoute(route); err != nil {
+		if err := createRoute(r.db, r.router, app.ID, route); err != nil {
```

Comment:
We should move this logic to after committing the transaction, so if the transaction fails we don't have a dangling route.